lib/delayed/backend/active_record.rb

```diff
@@ -137,11 +137,16 @@ def self.jobs_count(flavor,
         # to specify the jobs to act on, either pass opts[:ids] = [list of job ids]
         # or opts[:flavor] = <some flavor> to perform on all jobs of that flavor
         def self.bulk_update(action, opts)
-          scope = if opts[:flavor]
-            raise("Can't bulk update failed jobs") if opts[:flavor].to_s == 'failed'
+          raise("Can't #{action.to_s} failed jobs") if opts[:flavor].to_s == 'failed' && action.to_s != 'destroy'
+          scope = if opts[:ids]
+            if opts[:flavor].present? && opts[:flavor] == 'failed'
```

Comment:
there's really no need for the `ops[:flavor].present?` - it will either be `nil` or `''`, both of which will return false when comparing against `'failed'` (another type may pass the `present?` check, but raise on comparison to `'failed'`)